ttnn/cpp/ttnn/operations/conv/conv_transpose2d/conv_transpose2d.cpp

```diff
@@ -0,0 +1,350 @@
+                transform_weights_for_conv_transpose2d(weight_tensor),
+                bias_tensor,
+                conv_config.input_channels_alignment,
+                conv_config.weights_dtype,
+                opt_conv_op_block_config.act_block_w_ntiles,
+                opt_conv_op_block_config.out_subblock_w_ntiles,
+                parallel_config,
+                device,
+                groups,
+                opt_conv_op_block_config.act_block_h_ntiles,
+                input_width);
+        }
+        // call conv micro op
+        auto conv_output = optimized_conv_new(
+            halo_output,
+            weight_tensor_on_device,
+            bias_tensor_on_device,
+            sliding_window_config,
+            out_channels,
+            groups,
+            conv_config.output_layout == Layout::ROW_MAJOR,
+            conv_config.activation == "relu",
+            conv_config.math_fidelity,
+            opt_conv_op_parallel_config,
+            opt_conv_op_block_config,
+            conv_out_memory_config,
+            conv_config.dtype,
+            {batch_size, input_height, input_width, in_channels},
+            conv_config.input_channels_alignment == 16,
+            compute_kernel_config,
+            conv_config.enable_act_double_buffer,
+            conv_config.enable_split_reader,
+            conv_config.enable_subblock_padding);
+        ttnn::operations::core::deallocate(halo_output);
+        return {conv_output, output_height, output_width, weight_tensor_on_device, bias_tensor_on_device};
+    }
+
+template std::tuple<ttnn::Tensor, uint32_t, uint32_t, ttnn::Tensor, std::optional<ttnn::Tensor>> conv_transpose2d<Device>(
+    const ttnn::Tensor& input_tensor,
+    const ttnn::Tensor& weight_tensor,
+    Device * device,
+    uint32_t in_channels,
+    uint32_t out_channels,
+    uint32_t batch_size,
+    uint32_t input_height,
+    uint32_t input_width,
+    std::array<uint32_t, 2> kernel_size,
+    std::array<uint32_t, 2> stride,
+    std::array<uint32_t, 2> padding,
+    std::array<uint32_t, 2> output_padding,
+    std::array<uint32_t, 2> dilation,
+    uint32_t groups,
+    std::optional<const ttnn::Tensor> bias_tensor,
+    std::optional<const conv2d::Conv2dConfig> conv_config_);
+
+template std::tuple<ttnn::Tensor, uint32_t, uint32_t, ttnn::Tensor, std::optional<ttnn::Tensor>> conv_transpose2d<MeshDevice>(
+    const ttnn::Tensor& input_tensor,
+    const ttnn::Tensor& weight_tensor,
+    MeshDevice * device,
+    uint32_t in_channels,
+    uint32_t out_channels,
+    uint32_t batch_size,
+    uint32_t input_height,
+    uint32_t input_width,
+    std::array<uint32_t, 2> kernel_size,
+    std::array<uint32_t, 2> stride,
+    std::array<uint32_t, 2> padding,
+    std::array<uint32_t, 2> output_padding,
+    std::array<uint32_t, 2> dilation,
+    uint32_t groups,
+    std::optional<const ttnn::Tensor> bias_tensor,
+    std::optional<const conv2d::Conv2dConfig> conv_config_);
```

Comment:
Please remove template instantiation. It is not necessary.

Reply:
Removed.